client/src/main/java/io/pravega/client/stream/impl/EventStreamReaderImpl.java

```diff
@@ -226,7 +229,7 @@ private void handleSegmentTruncated(SegmentInputStream segmentReader) throws Rei
             long startingOffset = metadataClient.getSegmentInfo().getStartingOffset();
             segmentReader.setOffset(startingOffset);
         } catch (NoSuchSegmentException e) {
-            handleEndOfSegment(segmentReader);
+            handleEndOfSegment(segmentReader, true);
```

Comment:
The value 'true' passed here may or may not be correct. IE: If the segment is truncated away completely, does not mean it was not the last segment to be read.

Reply:
Successors are fetched here as we are passing true to handleEndOfSegment.